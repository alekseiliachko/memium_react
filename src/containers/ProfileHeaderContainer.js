import { connect } from "react-redux";
import { ProfileHeader } from "../components/ProfileHeader";
import {
  loadDetails,
  loadSubs,
  loadAvatar,
  updateDetails,
} from "../redux/user/actions";

const mapStateToProps = (store) => ({
  details: store.userReducer.details,
  avatar: store.userReducer.avatar,
  subs: store.userReducer.subs,
  avatarLoadingStatus: store.userReducer.avatarLoadingStatus,
  detailsLoadingStatus: store.userReducer.detailsLoadingStatus,
  subsLoadingStatus: store.userReducer.subsLoadingStatus,
});

const mapDispatchToProps = (dispatch) => ({
  loadDetails: () => dispatch(loadDetails()),
  loadAvatar: () => dispatch(loadAvatar()),
  loadSubs: () => dispatch(loadSubs()),
  updateDetails: (details) => dispatch(updateDetails(details)),
});

export const ProfileHeaderContainer = connect(
  mapStateToProps,
  mapDispatchToProps
)(ProfileHeader);
