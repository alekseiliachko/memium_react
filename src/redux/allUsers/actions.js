import AccountOpenController from "../../api/AccountOpenController";

export const SET_USER_DATA = "SET_USER_DATA";
export const setUserData = (userId, userData) => ({
  type: SET_USER_DATA,
  userId,
  userData,
});

export const fetchUserData = (userId) => async (dispatch) => {
  const res = await AccountOpenController.getAccountsDetails(userId);
  dispatch(setUserData(userId, res.data));
};

export const SET_USER_AVATAR = "SET_USER_AVATAR";
export const setUserAvatar = (userId, avatar) => ({
  type: SET_USER_AVATAR,
  userId,
  avatar,
});
export const fetchUserAvatar = (userId) => async (dispatch) => {
  const res = await AccountOpenController.getAccountsAvatar(userId);
  dispatch(setUserAvatar(userId, res));
};
