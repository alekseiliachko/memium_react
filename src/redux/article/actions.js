import ArticlesController from "../../api/ArticlesController";

export const SET_ARTICLE_DATA = "SET_ARTICLE_DATA";

export const setArticleData = (articleId, articleData) => ({
  type: SET_ARTICLE_DATA,
  articleId,
  articleData,
});

export const fetchArticleData = (articleId) => async (dispatch) => {
  const res = await ArticlesController.getArticleById(articleId);
  dispatch(setArticleData(articleId, res.data));
};
