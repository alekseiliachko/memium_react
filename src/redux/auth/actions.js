import AuthController from "../../api/AuthController";

export const LOGIN_ATTEMPT = "LOGIN_ATTEMPT";
export const LOGIN_SUCCESS = "LOGIN_SUCCESS";
export const LOG_OUT = "LOG_OUT";

export const SIGNUP_ATTEMPT = "SIGNUP_ATTEMPT";
export const SIGNUP_SUCCESS = "SIGNUP_SUCCESS";

export const AUTH_FAILURE = "AUTH_FAILURE";

export const setAuthStatusAttempt = () => ({
  type: LOGIN_ATTEMPT,
});

export const setAuthStatusSuccess = (username) => ({
  type: LOGIN_SUCCESS,
  payload: username,
});

export const logout = () => async (dispatch) => {
  localStorage.removeItem("memium_token");
  localStorage.removeItem("name");
  return {
    type: LOG_OUT,
  };
};

export const authFailure = (err) => ({
  type: AUTH_FAILURE,
  payload: err,
});

export const signupSuccess = () => ({
  type: SIGNUP_SUCCESS,
});

export const restoreSessionAttempt = () => (dispatch) => {
  dispatch(setAuthStatusAttempt());
  if (localStorage.memium_token) {
    AuthController.updateAuthHeader(localStorage.memium_token);
    dispatch(setAuthStatusSuccess(localStorage.name));
  } else {
    dispatch(logout());
  }
};

export const loginAttempt = (userInfo) => async (dispatch) => {
  dispatch(setAuthStatusAttempt());
  const res = await AuthController.login(userInfo);
  AuthController.updateAuthHeader(res.data.token);
  localStorage.setItem("name", res.data.username);
  dispatch(setAuthStatusSuccess(res.data.username));
};

export const signupAttempt = (userInfo) => async (dispatch) => {
  const res = AuthController.signup(userInfo);
  dispatch(signupSuccess());
};
