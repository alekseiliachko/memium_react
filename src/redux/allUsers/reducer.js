import { SET_USER_AVATAR, SET_USER_DATA } from "./actions";

export const initialState = {
  userDataById: {},
};

export const allUsersReducer = (store = initialState, action) => {
  switch (action.type) {
    case SET_USER_DATA:
      return {
        ...store,
        userDataById: {
          ...store.userDataById,
          [action.userId]: {
            ...store.userDataById[action.userId],
            ...action.userData,
          },
        },
      };
    case SET_USER_AVATAR:
      return {
        ...store,
        userDataById: {
          ...store.userDataById,
          [action.userId]: {
            ...store.userDataById[action.userId],
            avatar: action.avatar,
          },
        },
      };
    default:
      return store;
  }
};
