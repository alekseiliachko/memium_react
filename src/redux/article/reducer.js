import { RESET_ARTICLE_DATA, SET_ARTICLE_DATA } from "./actions";

const initialState = {
  articleDataById: {},
};

export const articleReducer = (store = initialState, action) => {
  switch (action.type) {
    case SET_ARTICLE_DATA:
      return {
        ...store,
        articleDataById: {
          ...store.articleDataById,
          [action.articleId]: action.articleData,
        },
      };
    case RESET_ARTICLE_DATA:
      return {
        ...store,
        articleDataById: {
          ...store.articleDataById,
          [action.articleId]: undefined,
        },
      };
    default:
      return store;
  }
};
