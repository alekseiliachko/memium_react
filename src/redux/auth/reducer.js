import { LOGIN_ATTEMPT, LOGIN_SUCCESS, LOG_OUT } from "./actions";

export const AUTH_STATE = {
  LOGGED_OUT: "LOGGED_OUT",
  SUCCESS: "SUCCESS",
  ATTEMPT: "ATTEMPT",
};

const initialState = {
  username: "",
  authState: AUTH_STATE.ATTEMPT,
};

export const authReducer = (store = initialState, action) => {
  switch (action.type) {
    case LOGIN_ATTEMPT:
      return {
        ...store,
        authState: AUTH_STATE.ATTEMPT,
      };
    case LOGIN_SUCCESS:
      return {
        ...store,
        username: action.payload,
        authState: AUTH_STATE.SUCCESS,
      };
    case LOG_OUT:
      return {
        ...store,
        username: "",
        authState: AUTH_STATE.LOGGED_OUT,
      };
    default:
      return store;
  }
};
