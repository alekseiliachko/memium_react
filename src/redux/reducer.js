import { combineReducers, createStore, applyMiddleware, compose } from "redux";
import thunk from "redux-thunk";
import { authReducer } from "./auth/reducer";
import { userReducer } from "./user/reducer";
import { articleReducer } from "./article/reducer";
import { allUsersReducer } from "./allUsers/reducer";
import { searchReducer } from "./search/reducer";

const rootState = combineReducers({
  authReducer,
  userReducer,
  articleReducer,
  allUsersReducer,
  searchReducer,
});

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;
const middleware = [thunk];
export function configureStore() {
  const store = createStore(
    rootState,
    composeEnhancers(applyMiddleware(...middleware))
  );
  return store;
}
