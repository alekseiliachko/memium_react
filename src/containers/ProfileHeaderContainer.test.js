import React from "react";
import renderer from "react-test-renderer";
import { withStore } from "../redux/testHelper";
import { MemoryRouter } from "react-router";
import { ProfileHeaderContainer } from "./ProfileHeaderContainer";
import { loadAvatar, loadDetails, loadSubs } from "../redux/user/actions";
import { LOADING_STATUS } from "../redux/user/reducer";

jest.mock("../redux/user/actions", () => {
  return {
    loadDetails: jest.fn(),
    loadAvatar: jest.fn(),
    loadSubs: jest.fn(),
  };
});

describe("profile header container", () => {
  it("renders correctly with data ready", async () => {
    const Comp = withStore();
    let tree;

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileHeaderContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it("renders correctly without data", async () => {
    const Comp = withStore({
      userReducer: {
        avatar: "",
        blackList: [],
        details: {
          bio: "",
          gender: "Another",
          name: "",
        },
        subs: [],
        likedList: [],
        avatarLoadingStatus: LOADING_STATUS.NOT_LOADED,
        detailsLoadingStatus: LOADING_STATUS.NOT_LOADED,
        subsLoadingStatus: LOADING_STATUS.NOT_LOADED,
        blackListLoadingStatus: LOADING_STATUS.NOT_LOADED,
        likedListLoadingStatus: LOADING_STATUS.NOT_LOADED,
        feedLoadingStatus: LOADING_STATUS.NOT_LOADED,
        feed: {},
        myArticles: [],
        myArticlesLoadingStatus: LOADING_STATUS.NOT_LOADED,
      },
    });
    let tree;

    loadAvatar.mockReturnValue({ type: "who_cares" });
    loadDetails.mockReturnValue({ type: "who_cares" });
    loadSubs.mockReturnValue({ type: "who_cares" });

    await renderer.act(async () => {
      tree = renderer.create(
        <Comp>
          <MemoryRouter>
            <ProfileHeaderContainer />
          </MemoryRouter>
        </Comp>
      );
      await Promise.resolve();
    });

    expect(tree.toJSON()).toMatchSnapshot();
    expect(loadAvatar).toBeCalled();
  });
});
