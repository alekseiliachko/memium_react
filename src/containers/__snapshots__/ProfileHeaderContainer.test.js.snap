// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`profile header container renders correctly with data ready 1`] = `
<div
  className="profile-header"
>
  <div
    className="MuiBox-root MuiBox-root-6"
  >
    <div
      className="MuiBox-root MuiBox-root-8 profile-header__details"
    >
      <div
        className="MuiBox-root MuiBox-root-9"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          Круто чел
        </h4>
        <button
          aria-label="bookmark"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <h6
        className="MuiTypography-root details__about MuiTypography-subtitle1"
      >
        populated
      </h6>
      <a
        href="/profile?tab=2"
        onClick={[Function]}
      >
        <p
          className="MuiTypography-root makeStyles-subs-7 MuiTypography-body1"
        >
          1
           подписок
        </p>
      </a>
    </div>
    <div
      className="MuiBox-root MuiBox-root-10 profile-header__avatar"
    >
      <input
        accept="image/*"
        className="makeStyles-input-3"
        onChange={[Function]}
        type="file"
      />
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-avatatLoader-4"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-1"
          >
            <img
              className="MuiAvatar-img"
              src="blob:http://localhost:3000/3dcb8dbb-23c7-4abf-8958-52663e0013da"
            />
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-5"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`profile header container renders correctly without data 1`] = `
<div
  className="profile-header"
>
  <div
    className="MuiBox-root MuiBox-root-11"
  >
    <div
      className="MuiBox-root MuiBox-root-12 profile-header__details"
    >
      <div
        className="MuiBox-root MuiBox-root-13"
      >
        <h4
          className="MuiTypography-root MuiTypography-h4"
        >
          
        </h4>
        <button
          aria-label="bookmark"
          className="MuiButtonBase-root MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <span
            className="MuiIconButton-label"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 17.25V21h3.75L17.81 9.94l-3.75-3.75L3 17.25zM20.71 7.04c.39-.39.39-1.02 0-1.41l-2.34-2.34a.9959.9959 0 00-1.41 0l-1.83 1.83 3.75 3.75 1.83-1.83z"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root"
          />
        </button>
      </div>
      <h6
        className="MuiTypography-root details__about MuiTypography-subtitle1"
      >
        
      </h6>
      <a
        href="/profile?tab=2"
        onClick={[Function]}
      >
        <p
          className="MuiTypography-root makeStyles-subs-7 MuiTypography-body1"
        >
          0
           подписок
        </p>
      </a>
    </div>
    <div
      className="MuiBox-root MuiBox-root-14 profile-header__avatar"
    >
      <input
        accept="image/*"
        className="makeStyles-input-3"
        onChange={[Function]}
        type="file"
      />
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-avatatLoader-4"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <div
            className="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-1 MuiAvatar-colorDefault"
          >
            <svg
              aria-hidden={true}
              className="MuiSvgIcon-root MuiAvatar-fallback"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 12c2.21 0 4-1.79 4-4s-1.79-4-4-4-4 1.79-4 4 1.79 4 4 4zm0 2c-2.67 0-8 1.34-8 4v2h16v-2c0-2.66-5.33-4-8-4z"
              />
            </svg>
          </div>
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root makeStyles-icon-5"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 4V1h2v3h3v2H5v3H3V6H0V4h3zm3 6V7h3V4h7l1.83 2H21c1.1 0 2 .9 2 2v12c0 1.1-.9 2-2 2H5c-1.1 0-2-.9-2-2V10h3zm7 9c2.76 0 5-2.24 5-5s-2.24-5-5-5-5 2.24-5 5 2.24 5 5 5zm-3.2-5c0 1.77 1.43 3.2 3.2 3.2s3.2-1.43 3.2-3.2-1.43-3.2-3.2-3.2-3.2 1.43-3.2 3.2z"
            />
          </svg>
        </span>
        <span
          className="MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</div>
`;
